### ISSUE TITLE
MediaStoreData putObject fails in browser due to preflight

### ISSUE THREAD

**Issue by zacharynevin** [aws/aws-sdk-js#1827]
Hi there,

### Description of use case

I have the following usecase for AWS MediaStore:

- Users record videos in the browser. These videos can be anywhere from 10 minutes to several hours. 
- While the video is being recorded, the intention is that the blobs are uploaded _directly_ to AWS MediaStore.

### What I did 

First, I call a method on my server to generate a federated token with a very limited set of permissions (i.e. MediaStoreData `putObject` for the upload path only), and send these temporary credentials down to the user.

Then, I take those credentials and use them to create a new `MediaStoreData` object. This is shown below:

```javascript
const mediaStoreData = new MediaStoreData({
  region: 'us-east-1',
  accessKeyId: <redacted>,
  secretAccessKey: <redacted>,
  sessionToken: <redacted>,
  endpoint: "https://i5domqqdltrwci.data.mediastore.us-east-1.amazonaws.com",
  httpOptions: {
    xhrWithCredentials: true
  }
})
```

When a new blob is available from the recorder, I send it to MediaStore using `putObject`:

```javascript
recorder.ondataavailable = blob => {
  mediaStoreData.putObject({
    Body: blob,
    Path: "recordings/myrecording.webm",
    ContentType: "video/webm",
    StorageClass: "TEMPORAL"
  })
}
```

### Description of error

When I make the `putObject `request, the browser makes an `OPTIONS` preflight request. This is where it fails. The `OPTIONS` request fails with a `403` error and the following message:
```xml
<AccessDeniedException>
  <Message>Unable to determine service/operation name to be authorized</Message>
</AccessDeniedException>
```

### Additional information

- I am currently testing this on `localhost`.

**Comment by chrisradek** [comment]
@zacharynevin 
Thanks for reporting this! It appears this service does not currently support CORS. I've forwarded this request (and https://forums.aws.amazon.com/thread.jspa?threadID=268939&tstart=0) to the Elemental team to prioritize, but don't have any ETA on when this will be completed.

I have tested and confirmed it works in node.js, so you can forward the blob to a Lambda function/server as a work-around until this support is added.

**Comment by zacharynevin** [comment]
Thanks @chrisradek!

I remember reading somewhere in the documentation that the `endpoint` parameter in `AWS.Config` is of the form `https://<service_name>.<region>.amazonaws.com`. If the authorizer can't deal with endpoints from `MediaStore` (e.g. `https://i5domqqdltrwci.data.mediastore.us-east-1.amazonaws.com`), is there another query or header property I can add to the request to force it?

**Comment by chrisradek** [comment]
Unfortunately, I don't believe there is a way to work-around this aside from running in an environment that doesn't enforce CORS. Keep an eye on the forum post you opened, and I'll follow-up if I hear anything from them as well. For now you would need to send the blob somewhere that doesn't enforce CORS, and send the data to MediaStore from there.

**Comment by zacharynevin** [comment]
Alright I can do that. Thanks @chrisradek!

**Comment by srchase** [comment]
A change to make this available in the browser build has been merged.  It will be available in the next release: v2.382.0
